Sources/TSCUtility/Version.swift

```diff
@@ -119,11 +120,15 @@ public extension Version {
             .split(separator: ".", maxSplits: 2, omittingEmptySubsequences: false)
             .map(String.init).compactMap({ Int($0) }).filter({ $0 >= 0 })
 
-        guard requiredComponents.count == 3 else { return nil }
+        guard requiredComponents.count == 3 || (requiredComponents.count == 2 && lenient) else { return nil }
```

Comment:
If we're doing this, would it be better (or make sense) if the rules are made even more lenient to allow for just a single major version number? 

i.e.

```swift
guard requiredComponents.count == 3 || lenient else ...
```

Reply:
My original intention was to make repositories that mix two and three component version styles more compatible. Supporting a single version component would make SPM support repositories that don't technically version semantically, which is a larger change. But that behavior can still map to semantic versions, and it would maximize compatibility with existing repos.

Reply:
I tend to agree that from a conversion point of, tags are either 3 `x.0.0` or 2 `x.0` segment long. iow a single digit version tag would be pretty exceptional.